src/main/java/io/vertx/core/http/impl/VertxHttp2ConnectionHandler.java

```diff
@@ -397,6 +396,22 @@ public void onSettingsRead(ChannelHandlerContext ctx, Http2Settings settings) th
     }
   }
 
+  @Override
+  public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+    if (msg instanceof Http2StreamFrame) {
+      // Handle HTTP/2 clear text upgrade request
```

Comment:
I don't get this comment. If we have an `Http2StreamFrame`, isn't the connection already upgraded?

Reply:
that's the frame actually sent by `Http2UpgradeHandler` which are translated from HTTP/2 `HttpObject`, e.g

```
            HttpContent content = (HttpContent) msg;
            ByteBuf buf = VertxHandler.safeBuffer(content.content(), ctx.alloc());
            boolean end = msg instanceof LastHttpContent;
            ctx.fireChannelRead(new DefaultHttp2DataFrame(buf, end, 0));
```

that is in `Http2UpgradeHandler`

Reply:
normally the `VertxHttp2ConnectionHandler` expects `ByteBuf` that it will transform into such frames, during the upgrade we directly receive these frames from the upgrade handler